### ISSUE TITLE
Make "loop single" playback order utilize .adx loop points?

### ISSUE THREAD

**Issue by Soukyuu** [jchv/deadbeef-vgmstream#1]
Thanks for making (porting?) this plugin to the deadbeef world. foobar2000's version has an option to loop the .adx files containing a loop point if playback order is set to "repeat", thus allowing seamless looping of a track just as it does in the game.

Would be really nice if you could implement this as well.

**Comment by jchv** [comment]
Hi. Glad to hear you like it. It was written from scratch, but given that all it does is interface between deadbeef and vgmstream, it was probably easier than porting anyway.

I do not use Linux as my primary platform and so have not touched this in a while. However, I will note that originally I wanted this functionality, too. Unfortunately, I think the only way I can make it work with current Deadbeef would be to have an option that makes the plugin make all loopable files 'endless' - which would work, but would require you to switch an option every time you want to repeat single on files.

I guess my best bet is to check out the activity on Deadbeef and make sure this is the case, so I'm going to do that and see what I find...

**Comment by jchv** [comment]
I don't think anything has changed on this unfortunately. Guess I will have to implement it by adding an option in the plugin.

**Comment by Soukyuu** [comment]
Can't you do something like [ddb_replaygain_control plugin](https://github.com/cboxdoerfer/ddb_replaygain_control/blob/master/rg_control.c#L427) does to query current playback mode? Just query `playback.mode` instead of `playback.order` there.

**Comment by jchv** [comment]
Yeah, looks like I could do that. Thanks for pointing that out. I'll take a look after work.

**Comment by Soukyuu** [comment]
Did you ever get to look at this one?

**Comment by jchv** [comment]
Yes, I actually did look into it, and I found it is impossible to do because length is calculated upon adding music to the playlist, not upon playing it. As far as I know, there's not really a good way of doing this in Deadbeef. It would be possible to just add a loop single option for the _plugin_ that makes all tracks last forever, and that seems like all that can be done. (I haven't yet added anything like that, though.)

**Comment by jchv** [comment]
I never closed this issue, but I did actually fix it some time ago. If you use loop single, it will look like it's going to end and fade out, but instead it will just continue playing indefinitely past the "end".
